packages/react/primitive/src/Primitive.tsx

```diff
@@ -3,32 +3,32 @@ import { Slot } from '@radix-ui/react-slot';
 
 const NODES = ['a', 'button', 'div', 'h2', 'h3', 'p', 'img', 'span', 'svg'] as const;
 
-type MergeProps<P1 = {}, P2 = {}> = Omit<P1, keyof P2> & P2;
-type Primitives = { [E in typeof NODES[number]]: PrimitiveForwardRefComponent<E> };
-
-type PrimitiveForwardRefComponent<E extends React.ElementType> = React.ForwardRefExoticComponent<
-  MergeProps<React.ComponentPropsWithRef<E>, PrimitiveProps>
->;
-
 // Temporary while we await merge of this fix:
 // https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55396
-type PropsWithoutRef<P> = P extends { ref?: any } ? Pick<P, Exclude<keyof P, 'ref'>> : P;
+// prettier-ignore
+type PropsWithoutRef<P> = P extends any ? ('ref' extends keyof P ? Pick<P, Exclude<keyof P, 'ref'>> : P) : P;
```

Comment:
This has changed a little, would you mind explaining the changes roughly?

Reply:
Ah yeah, so... the original typedef that `DefinitelyTyped` had ensured that there was no unnecessary `Pick`ing over things by checking if `'ref' extends keyof P` but my type would still `Pick` if `P` was an empty object, so you'd end up with `Pick<{}, Exclude<keyof {}, 'ref'>>`.

I've changed it now so that `P extends any` activates the distributive conditional over the union and then applies the same original logic that the `DefinitelyTyped` types had. Hopefully this is more likely to be approved sooner by them too as it is less of a change to their original defs.